File: Python/kadaneAlgo.py
from sys import maxsize

def maxSubArrSum(arr, size):
    start = 0 #Starting index
    end = 0 #Ending index
    s = 0
    max_so_far = -maxsize -1
    max_ending_here = 0
    for i in range(size):
        max_ending_here += arr[i]
        if (max_so_far < max_ending_here):
            max_so_far=max_ending_here 
            start = s
            end = i
        if (max_ending_here<0):
            max_ending_here = 0
            s = i+1
    print("Starting Index %d" % (start))
    print("Ending Index %d" % (end))
    return max_so_far

arr = [-2,-3,4,-1,-2,1,5,-3]
print("Given Array:",arr)
size = len(arr)

max_sum = maxSubArrSum(arr, len(arr))

print("Maximum Contiguous Sum:", max_sum)

File: C++/kadaneAlgo.cpp
//The task is to find the sum of the contiguous subarray within a arr[] with the largest sum.
#include<iostream>
using namespace std;

int maxSubArraySum(int arr[], int size) {
    int start = 0;
    int end = 0;
    int s = 0;
    int max_so_far = INT8_MIN;
    int max_ending_here = 0;
    for(int i=0; i<size; i++){
        max_ending_here += arr[i];
        if (max_so_far<max_ending_here) {
            max_so_far = max_ending_here;
            start = s;
            end = i;
        } 
        if(max_ending_here<0) {
            max_ending_here = 0;
            s = i+1;
        }
    }
    cout<<"Starting index: "<<start<<endl;
    cout<<"Ending index: "<<end<<endl;
    return max_so_far;
}

int main() {
    int arr[] = {-2,-3,4,-1,-2,1,5,-3};
    int size = sizeof(arr) / sizeof(arr[0]);
    int max_sum = maxSubArraySum(arr, size);
    cout<<"Maximum Contiguous Sum: "<<max_sum;
    return 0;
}